azkaban-common/src/main/java/azkaban/server/AzkabanServer.java

```diff
@@ -71,11 +85,41 @@ public static Props loadProps(final String[] args, final OptionParser parser) {
         azkabanSettings = loadAzkabanConfigurationFromDirectory(dir);
       }
     } else {
-      logger
-          .info("Conf parameter not set, attempting to get value from AZKABAN_HOME env.");
+      logger.info("Conf parameter not set, attempting to get value from AZKABAN_HOME env.");
       azkabanSettings = loadConfigurationFromAzkabanHome();
     }
 
+    try {
+      if (options.has(emailDirectory)) {
+        final String path = options.valueOf(emailDirectory);
+        logger.info("loading email templates from path " + path);
+        final File dir = new File(path);
+        if (!dir.exists()) {
+          logger.error("email templates: path " + path + " does not exist.");
+        } else if (!dir.isDirectory()) {
+          logger.error("email templates: path " + path + " is not a directory.");
+        } else {
+          MailCreatorRegistry.registerCreator(TemplateBasedMailCreator.fromPath(path), true);
+        }
+      } else {
+        final String azkabanHome = System.getenv("AZKABAN_HOME");
+        final File emailPath = new File(azkabanHome, Constants.DEFAULT_EMAIL_TEMPLATE_PATH);
```

Comment:
I think it's better to check azkabanHome first. If it's null, then we don't create the new emailPath.

Reply:
I doubt whether AZKABAN_HOME will be used since conf and email should always be passed in the argument list. If that's the case, I don't even think we need to check AZKABAN_HOME.

Reply:
fair point, adapting the code to what you are proposing